src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToStringTransformer.php

```diff
@@ -132,6 +132,14 @@ public function reverseTransform($value)
             throw new TransformationFailedException('Expected a string.');
         }
 
+        // handle seconds ignored by user's browser when seconds as single_text is 0
+        if ($this->parseFormat === 'H:i:s|') {
+            if (!preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9])(?::[0-5][0-9]))', $value) &&
+                preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9]))', $value)) {
```

Comment:
> Here just we managed the browser omission.

So this doesnt count for minutes and hours then? Sorry no real time to test this. But i proposed are more or less generic approach for "adding sensible defaults to the value". Ie.

```
format="H", value="", new_value="00"
format="H:i", value="", new_value="00:00"
format="H:i", value="12", new_value="12:00"
etc.
```